### PR TITLE
relay-runtime: Add types for readUpdatableFragment_EXPERIMENTAL

### PR DESCRIPTION
**Pull request by v-morlock**
This functionality was partially typed already and seems to be fully compatible with typescript

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [Documented here](https://relay.dev/docs/guided-tour/updating-data/imperatively-modifying-store-data/)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/relay-runtime/lib/store/RelayStoreTypes.d.ts [hunk 1]
```diff
@@ -1041,15 +1046,16 @@ export type RelayResolverErrors = RelayResolverError[];
...
  * A linked field where an updatable fragment is spread has the type
  * HasUpdatableSpread.
  * This type is expected by store.readUpdatableFragment_EXPERIMENTAL.
  */
-export interface HasUpdatableSpread<TFragmentType> {
-    readonly $updatableFragmentSpreads: TFragmentType;
-}
+export type HasUpdatableSpread<TData = unknown> = Readonly<{
+    ' $data'?: TData | undefined;
+    $updatableFragmentSpreads: FragmentType;
```
Comment (maraisr) [inline types/relay-runtime/lib/store/RelayStoreTypes.d.ts:1060]: probably something bigger at play to add this new field in all the places.

File: types/relay-runtime/relay-runtime-tests.tsx [hunk 2]
```diff
@@ -1,28 +1,12 @@
 import {
```
Comment (maraisr) [inline types/relay-runtime/relay-runtime-tests.tsx:1]: do you mind resetting this import block, hard to see what's changed.

**Review by maraisr — APPROVED** [review]
few comments if you choose to address, but all good if not. 

think we should circle back at some point and look at the new `-experimental` package Relay has splintered anyway.

But thanks! 🚀

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## relay-runtime ([<kbd>unpkg</kbd>](https://unpkg.com/browse/relay-runtime@latest/))
was missing the following properties:
1. MutationTypes
1. RangeOperations
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a3c4720715dbb9f07fc6dfb46377c6c282ec8ff9
</p>

**Review by maraisr — APPROVED** [review]
<3 thank you so much! ship it!

**Comment by v-morlock** [comment]
ready to merge

**Pull request merged**